### PR TITLE
Use abstract base class for container operations

### PR DESCRIPTION
**Pull request by willcl-ark**

### CODE REVIEW

File: src/warnet/tank.py [hunk 1]
```diff
@@ -116,30 +124,37 @@ def ipv4(self):
...
             )
         return self._exporter_name
 
     @property
     def container(self) -> Container:
+        # cast makes typecheckers happy
         if self._container is None:
             try:
-                self._container = docker.from_env().containers.get(self.container_name)
+                self._container = cast(Container, docker.from_env().containers.get(self.container_name))
```
Comment (pinheadmz) [inline src/warnet/tank.py]: I was grepping for leftover `docker.from_env()` calls and I just wanna make sure that this line here is still abstractable

**Comment by willcl-ark** [comment]
OK going to squash from 9057cef back

**Comment by willcl-ark** [comment]
I didn't address the extra `services` in this one, but @josibake services are all inside `DockerInterface` so you shouldn't be bothered by them while working on getting tanks running on k8s (although without tor running some tests might not work).

I can add services to this one, or do in a followup (I'd prefer followup I think, but not strongly).

**Comment by josibake** [comment]
LGTM, :shipit: 

> I can add services to this one, or do in a followup (I'd prefer followup I think, but not strongly).

followup :+1:

File: src/warnet/tank.py (hunk 1 above)
Comment (josibake) [reply src/warnet/tank.py]: nice catch, looks like this was fixed

**Pull request merged**